Add set and get commands for ESP RainMaker console (MEGH-6740)

Infishiv opened the pull request:
This PR adds two new console commands to ESP RainMaker:
     
     - `set`: Allows setting parameter values for devices
       Usage: set <device_name> <param_name> <value>
     
     - `get`: Allows getting parameter values from devices
       Usage: get <device_name> <param_name>
     
     These commands make it easier to interact with ESP RainMaker devices through the console interface.

Infishiv committed:
Add smart-set and smart-get commands for ESP RainMaker console

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/espressif/esp-rainmaker?pullRequest=358) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/espressif/esp-rainmaker?pullRequest=358) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/espressif/esp-rainmaker?pullRequest=358) it.</sub>

Infishiv committed:
Remove redundant includes and simplify command names

Infishiv committed:
fix: Address Copilot review feedback for set/get commands

- Update help strings with clear usage patterns
- Improve function naming consistency
- Add proper error handling and validation
- Implement type-safe value conversion
